drivers/gpu/drm/bridge/cdns-mhdp.c

```diff
@@ -1292,10 +1313,16 @@ static int mhdp_link_training(struct cdns_mhdp_device *mhdp,
 			      unsigned int training_interval)
 {
 	u32 reg32;
+	bool cr_done = false, eq_done = false;
+	int ret;
 	const u8 eq_tps = eq_training_pattern_supported(mhdp->host, mhdp->sink);
 
 	while (1) {
-		if (!mhdp_link_training_clock_recovery(mhdp)) {
+		ret = mhdp_link_training_clock_recovery(mhdp, &cr_done);
+		if (ret < 0)
+			goto err;
+
+		if (!cr_done) {
```

Comment:
I didn't go through the code, but this looks suspicious. Here you initialize cr_done to false, outside the while(1) loop. Then you pass it forward.

Shouldn't you either not need to initialize cr_done at all (expecting mhdp_link_training_clock_recovery to always set it), or initialize it every time before calling mhdp_link_training_clock_recovery()?

And similarly for eq_done.